go/common/ledger/ledger.go

```diff
@@ -6,9 +6,10 @@ import (
 )
 
 const (
-	// PathPurpose is set to 44 to indicate use of the BIP-0044 specification.
-	PathPurpose uint32 = 44
-
+	// PathPurposeEntity is set to 44 to indicate use of the BIP-0044 specification.
+	PathPurposeEntity uint32 = 44
+	// PathPurposeConsensus is set to 474 to be used with ledger Validator app
```

Comment:
Why is this set to `474`? Is there any unofficial agreement like [SLIP-0044](https://github.com/satoshilabs/slips/blob/master/slip-0044.md) for registered coin types for [BIP44](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki)?

Reply:
no, we want to completely separate the validator app from anything coin related

Reply:
> No, we want to completely separate the validator app from anything coin related.

Agreed.

> We can choose any other number for purpose if really necessary, but it is important that the path does not collide with normal BIP44 paths.

I agree it is important that this path doesn't collide with any other BIP44 paths. That's why I asked if there is any unofficial community agreement on how to "register" a purpose.

I actually found one proposal to augment BIP43 by a core Bitcoin developer which makes sense: https://github.com/bitcoin/bips/pull/523/.

It proposes non-Bitcoin cryptocoins should use:
```
<mnemonic> / 43' / <coin-type>' / <subpurpose>' /
```

where `<coin-type>` is the non-Bitcoin's coin type as registered in [SLIP-0044](https://github.com/satoshilabs/slips/blob/master/slip-0044.md) (`474` in our case) and `<subpurpose>` signifies the purpose, e.g. consensus, and would have a number that is unique in the Oasis' namespace. We free to choose whatever we like here.

Reply:
well... basically, if you don't start with 44' then you are in the realm of BIP32 where the structure is actually free.

Reply:
If this was arbitrarily chosen, I would prefer following the proposal I mentioned above and use:
```
<mnemonic> / 43' / 474' / <subpurpose>' /
```
where `<subpurpose>` is e.g. `0` for consensus.

Reply:
This break several parts of the code and integration libraries because we are already using 5 elements.

Reply:
I don't have a strong preference on how many path elements are used, just that they are prefixed with `43' / 474' / <subpurpose>' / `.

I tried to keep the number of elements at 5 in my suggested changes: https://github.com/oasisprotocol/oasis-core/pull/3058#discussion_r449126499.